packages/dds/matrix/src/matrix.ts

```diff
@@ -305,6 +363,7 @@ export class SharedMatrix<T = any>
 			localSeq,
 			rowsRefSeq,
 			colsRefSeq,
+			referenceSeqNumber: this.runtime.deltaManager.lastSequenceNumber,
```

Comment:
I'm a bit confused why we need to store this data in local metadata. I'd expect acked op to have all the data we need (including refSeq number), and thus resolution of the conflict to be the same for local & non-local ops.

I think this info is needed due to deficiency of our rebasing / resubmit flow. It does track original ref seq number, but does not provide one to the DDS when doing resubmission, and thus DDS has no clue. I've raised it before, but it did not matter for map / directory (as they have simple last writer wins policy that does not depend on actual ref seq number), and Sequence tracks it itself (probably through same flow as you are adding here).

If that's the case, we can leave it as is for now, but I think better long-term strategy would be to solve it once and for all by providing this info from pending state manager.

Reply:
Yes, I needed to add for the resubmission flow as you mentioned. For now, let's keep it as is and when we fix the pending state manager, then we can remove this. I will try to open a task for that so that it can scheduled sometime.,

Reply:
This makes sense. But storing clientId does not :) By the time we are asked to resubmit, old clientId would be useless, no?

Reply:
Yes, removed the Clientid. Thinking more about it, it is ok to remove ClientId, because it is resubmitting local ops only, so they are from this client only. So, the check is not required.